src/util/irep.cpp

```diff
@@ -346,6 +346,32 @@ irept &irept::add(const irep_namet &name, const irept &irep)
   #endif
 }
 
+#ifdef USE_MOVE
+irept &irept::add(const irep_namet &name, irept &&irep)
+{
+  named_subt &s = is_comment(name) ? get_comments() : get_named_sub();
+
+#ifdef SUB_IS_LIST
+  named_subt::iterator it = named_subt_lower_bound(s, name);
+
+  if(it == s.end() || it->first != name)
+    it = s.insert(it, std::make_pair(name, std::move(irep)));
+  else
+    it->second = std::move(irep);
+
+  return it->second;
+#else
+  std::pair<named_subt::iterator, bool> entry =
+    s.emplace(name, std::move(irep));
+
+  if(!entry.second)
+    entry.first->second = std::move(irep);
+
```

Comment:
Does the above work? I'd be worried that the first std::move gets evaluated, and then second call doesn't have anything to move anymore.

Reply:
No, it did not. I've got an implementation that does, will push soon - just working on PRs that set up proper use of all this.